problem-gson/src/test/java/org/zalando/problem/IOProblem.java

```diff
@@ -0,0 +1,56 @@
+package org.zalando.problem;
+
+import java.io.IOException;
+import java.net.URI;
+
+final class IOProblem extends IOException implements Exceptional {
+
+    private final URI type;
+    private final String title;
+    private final StatusType status;
+    private final String detail;
+    private final URI instance;
+
+    IOProblem(final URI type,
+              final String title,
+              final StatusType status,
+              final String detail,
+              final URI instance) {
+        this.type = type;
+        this.title = title;
+        this.status = status;
+        this.detail = detail;
+        this.instance = instance;
+    }
+
+    @Override
+    public URI getType() {
+        return type;
+    }
+
+    @Override
+    public String getTitle() {
+        return title;
+    }
+
+    @Override
+    public StatusType getStatus() {
+        return status;
+    }
+
+    @Override
+    public String getDetail() {
+        return detail;
+    }
+
+    @Override
+    public URI getInstance() {
+        return instance;
+    }
+
+    @Override
+    public ThrowableProblem getCause() {
```

Comment:
It's the same as the implementation in the super class, effectively. Can be removed.

Reply:
Yes effectively same as super class. I'll remove it.
I copied most of the test classes as is from jackson module - should we refactor those as well?

Reply:
It won't compile without it. Interface `Exceptional` requires `getCause()` to return a `ThrowableProblem`, where as the parent `IOException` returns `Throwable`.

Reply:
Ah, I see